src/extension.ts

```diff
@@ -434,23 +434,29 @@ function handleLogViaKubectlCli(podName: string) : Promise<string> {
 async function registerTutorialWithDidact(context: vscode.ExtensionContext) {
 	// call didact command to register tutorial
 	try {
-		// command ID: vscode.didact.register
-		const commandId = 'vscode.didact.register';
-
-		// then pass name, uri, and category
-		const tutorialName = 'Your First Integration';
-		const tutorialPath = path.join(context.extensionPath, './didact/camelk/first-integration.md');
-		const tutorialUri = vscode.Uri.parse(`file://${tutorialPath}`);
-		const tutorialCategory = 'Apache Camel K';
-
-		console.log('Tutorial URI registered: ' + tutorialUri.fsPath);
-
-		await vscode.commands.executeCommand(
-			commandId,
-			tutorialName, 
-			tutorialUri,
-			tutorialCategory);
-
+		// test to ensure didact is available 
+		const extensionId = 'redhat.vscode-didact';
+		const didactExt : any = vscode.extensions.getExtension(extensionId);
+
+		// if didact is available, register the new extension
+		if (didactExt) {
+			// command ID: vscode.didact.register
```

Comment:
this comment is useless, it is completely redundant with the code

Reply:
Since this is an exemplar, I would rather be explicit because some people will not be familiar with the VS Code command API.

Reply:
I don't understand why it is an exemplar.

Even if exemplar, providing comments that are duplicating exactly the code is giving no information to users. It is creating duplicating information that can go unsynchronized.

Reply:
> this comment is useless, it is completely redundant with the code

https://github.com/thoughtbot/guides/tree/master/code-review

Reply:
> Even if exemplar, providing comments that are duplicating exactly the code is giving no information to users. It is creating duplicating information that can go unsynchronized.

this statement is true for every comment written anywhere, and it's not an strong enough argument for not using comments.

"Clear enough code shouldn't require comments" is a very extreme position that developer can have but that shouldn't try to force on others.

Reply:
> "Clear enough code shouldn't require comments" is a very extreme position that developer can have but that shouldn't try to force on others.

in this case, this is not even this statement. it is for duplicating information. if the comment is providing more context, more information, why not. The pointed one is an exact duplication, there is no additional information provided.

Reply:
> > this comment is useless, it is completely redundant with the code
> 
> https://github.com/thoughtbot/guides/tree/master/code-review

sorry should formulate:

this comment is completely redundant with the code and doesn't provide any other piece of information. Given that it can be safely removed: no information will be lost and the code will be clearer.